### ISSUE TITLE
test: failing test_bind_carrier in systemd-networkd-tests.py

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#11724]
**systemd version the issue has been seen with**
> latest master

**Used distribution**
 > CentOS 7.6

`systemd-networkd-tests.py` started suddenly failing due to a recent change (see #11667), even though the CI run for the particular change has passed.

```
======================================================================
FAIL: test_bind_carrier (__main__.NetworkdNetWorkTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 996, in test_bind_carrier
    self.assertNotRegex(output, 'UP,LOWER_UP')
AssertionError: Regex matched: 'UP,LOWER_UP' matches 'UP,LOWER_UP' in '81: test1: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000\n    link/ether 76:6d:da:8b:bd:83 brd ff:ff:ff:ff:ff:ff\n    inet 192.168.10.30/24 brd 192.168.10.255 scope global test1\n       valid_lft forever preferred_lft forever\n    inet6 fe80::746d:daff:fe8b:bd83/64 scope link \n       valid_lft forever preferred_lft forever'
```
Full log: https://ci.centos.org/job/systemd-pr-build/4774/artifact/systemd-centos-ci/artifacts_h4qM7x/testsuite-logs.38f/systemd-networkd-tests.py_FAIL.log

This affects several recent PRs, so I'm labeling it as a priority.

cc @yuwata

**Comment by mrc0mmand** [comment]
Unfortunately, even with the fix from #11728 the test still fails:

```
======================================================================
FAIL: test_bind_carrier (__main__.NetworkdNetWorkTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./systemd-networkd-tests.py", line 993, in test_bind_carrier
    self.assertNotRegex(output, 'UP,LOWER_UP')
AssertionError: Regex matched: 'UP,LOWER_UP' matches 'UP,LOWER_UP' in '69: test1: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000\n    link/ether b2:ce:af:0b:e4:8b brd ff:ff:ff:ff:ff:ff\n    inet'

----------------------------------------------------------------------
Ran 72 tests in 799.244s

[root@n29 test-network]# git log -1 --oneline
63eb095 test-network: increase sleep time in test_bind_carrier()
```

I forgot to mention I temporarily disabled the systemd-networkd testsuite to keep results in other PRs relevant, thus the CentOS CI pass in #11728. Is there anything I could do to make the debugging easier for you, @yuwata?

**Comment by yuwata** [comment]
Ah, you already disabled the test. Sorry, I did not notice that. I will take a look.
